Frontend update: 2021-04 (#492)
SenseNet/sncom-resources: _updates/2021-04-19-april.md

---
title: April 2021
description: sensenet frontend updates of April 2021
---

# April 2021 🌸

"The Romans gave this month the Latin name Aprilis but the derivation of this name is uncertain. The traditional etymology is from the verb aperire, "to open", in allusion to its being the season when trees and flowers begin to "open", which is supported by comparison with the modern Greek use of άνοιξη (ánixi) (opening) for spring." /from Wikipedia/ 🌷💐

> For us not only the flowers and trees are opening, but the second quarter in this year is also started. We have so many plans for the Q2, we are really excited to share with you soon. We would like to renew our example apps and we also would like to add some new ones - the work has already started on this. We also changed the whole build process, so you have more options for using our packages now. 🔧

Reviewer: ```suggestion
For us not only the flowers and trees are blooming, but the second quarter has also begun. We have many plans for Q2 and we are really excited to share them with you soon. We would like to redesign our example apps and we also would like to add some new ones - the work has already started on this. We also updated the whole build process, so you have more options now for using our packages. 🔧
```

-----------

## admin-ui@1.19.0

We were also busy with fixing bugs in the last couple of weeks, e.g.: 🐛

[#1251](https://github.com/SenseNet/sn-client/pull/1251) ⏰
The end2end tests made it clear to us that a lot depends on the timing, so we added a loader to the 'Add button list' till it loads the new list.

[#1252](https://github.com/SenseNet/sn-client/pull/1252) 🚩

> We cleaned up the icons also, it is now more flexible, then ever before. You can select the icon what you would like to use from the CTD itself.

Reviewer: ```suggestion
We cleaned up type icons as well, now it is more flexible to use them, then ever before. You can select the type icon what you would like to display from the CTD itself.
```

[#1279](https://github.com/SenseNet/sn-client/pull/1279) ♻️

> Another main point what we are working on is the webhooks. We would like to create webhook templates to make your life easier with workflows.

Reviewer: ```suggestion
Another key point that we are working on is webhooks. We would like to create as many webhook templates as we can to make your life easier.
```

> Keep up with the next releases, it will be available soon, the first step for this was the editable payload.

Reviewer: ```suggestion
Keep up with the next releases, the templates will be available soon and the first step to reach them was the make webhook payload editable .
```

> ... and many more [#1248](https://github.com/SenseNet/sn-client/pull/1248), [#1250](https://github.com/SenseNet/sn-client/pull/1250).

Reviewer: ```suggestion
[#1248](https://github.com/SenseNet/sn-client/pull/1248), [#1250](https://github.com/SenseNet/sn-client/pull/1250).
```

## @sensenet/document-viewer-react@4.1.0

> In document viewer we had also some bugs or missing features what we tried to solve:

Reviewer: ```suggestion
In document viewer we had also some bugs or missing features what we did solve:
```

[#1259](https://github.com/SenseNet/sn-client/pull/1259) 📑

> Comment marker placing on inactive page has a bug, now it is working properly.

Reviewer: ```suggestion
There was a bug in comment marker placing on inactive page, now it is working properly.
```

[#1256](https://github.com/SenseNet/sn-client/pull/1256) 🔄

> One, not a very likely case is handled: if the backend cannot generate previews for a long time, it will not affect other opreations, infinite loop cannot be created.

Reviewer: ```suggestion
One edgecase is also has been adressed: if the backend cannot generate previews for a long time, it will not affect other opreations, infinite loop cannot be occured.
```

[#1260](https://github.com/SenseNet/sn-client/pull/1260) 🚫
Comment marker and shape placing at the same time is not available from now.

## @sensenet/universal-header-react@0.1.0

> Our packages have a newcomer: the universal header [#1231](https://github.com/SenseNet/sn-client/pull/1231) 💫

Reviewer: ```suggestion
A newcomer arrived in our packages: the universal header [#1231](https://github.com/SenseNet/sn-client/pull/1231) 💫
```

> Universal header was created to standardize our exmaple apps. This header can be used by anyone.

Reviewer: ```suggestion
Universal header was created to standardize our exmaple apps, but this header component can be used by anyone.
```

It has three main parts:

- Hamburger menu (with links of our main documentation and helper stuff) 🍔
- Title (can be overwritten)
- Three-dot menu (can be a custom menu for your app)

## @examples/gatsby-blog@0.1.0

> Our examples also have a newbie: gatsby-blog.

Reviewer: ```suggestion
There's a newbie example app as well: gatsby-blog.
```

> In this version it uses the sources from the file system, but the new source will be a sensenet repository soon. 😍

Reviewer: ```suggestion
In this version it uses the posts from the file system, but we are currently working on a source plugin so the source of this plugin will be a sensenet repository in the near future. 😍
```

-----------

You should also check the changes in the build system here: [#1234](https://github.com/SenseNet/sn-client/pull/1234)

> As you know we've been putting a lot of effort on optimization lately, so check out our article [here](https://dev.to/taki9/dynamic-import-recipe-for-a-lightning-fast-application-2fa0) about it.

Reviewer: ```suggestion
As you know we've been putting a lot of effort on optimization lately, check out our article [here](https://dev.to/taki9/dynamic-import-recipe-for-a-lightning-fast-application-2fa0) about it.
```
